### PR TITLE
Added provider AnimeBytes

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

Added backlog, haven't tested RSS yet.

* Needed to manually create the release names.
* Because there isn't any Season pack support for anime shows, i've added the Regex to the end of the anime regexes.
* Added a UnicodeDecodeError catch, because that's going wrong in NameParser() and where going to phase that out anyway.

Todo:
- [ ] Make sure it's not "accidentally" used for episode searches
- [ ] Check that I don't add items from results that where already added before.
- [ ] I don't know if that should be part of this PR, but maybe try to fix the RC for the UnicodeDecodeError exception.
- [ ] Test

### CODE REVIEW

**Comment by p0psicles** [comment]
Ab torrent urls change. So about every day you get a new url, meaning a double, tripple, ... searchresult. Need to think about that. Ideas?

**Comment by labrys** [comment]
Do the hashes change?

**Comment by p0psicles** [comment]
I don't know anything about torrents. So what do you mean with do the hashes change? Ik can share examples?

**Comment by duramato** [comment]
@labrys The urls don't have hashes, just torrent ids and user passkeys most likely in this format `SITE/torrents.php?action=download&id=73616&authkey=PASSKEY&torrent_pass=OTHERPASSKEY

**Comment by p0psicles** [comment]
Yes, and the id or another value, change after some time.

**Comment by duramato** [comment]
Most likely the passkey as the id is what id's the torrent on the tracker

**Comment by p0psicles** [comment]
It's something like this: https://animebytes.tv/torrents.php?action=download&id=111945&authkey=01DFSFEFWDFSDFSFSFSF&torrent_pass=DFASDWEDSDSS.

authkey changes.

**Comment by medariox** [comment]
@p0psicles 
Let the user add the authkey and save the url without it, then recreate the full url on snatch?

**Comment by p0psicles** [comment]
The only thing I can think of, is to connect back to AB when you try to snatch, to get the current URL. And just accept the double/tripple.. results in cache. Or set a short cache retention for this provider, making it drop results older then one day. Buy as these url's change frequently, you'l have to rescrape the torrent url.

File: sickrage/providers/GenericProvider.py [hunk 1]
```diff
@@ -180,7 +180,10 @@ def find_search_results(self, show, episodes, search_mode, forced_search=False,
             try:
                 parse_result = NameParser(parse_method=('normal', 'anime')[show.is_anime]).parse(title)
             except (InvalidNameException, InvalidShowException) as error:
-                logger.log(u"{}".format(error), logger.DEBUG)
+                logger.log(u"{error}".format(error=error), logger.DEBUG)
+                continue
+            except UnicodeDecodeError as error:
```
Comment (medariox) [inline sickrage/providers/GenericProvider.py]: Do we still need this?

Comment (p0psicles) [inline sickrage/providers/GenericProvider.py]: Good point. Lets remove it!

**Comment by p0psicles** [comment]
Did some more tests, searing, parsing and release creation goes well. But still medusa has some bugs, where it's not able to handle season packs / mulieps packs, icw scene namings. As AB is for 90% season packs / multieps / show packs, I think it will not always be as effective.

File: sickbeard/providers/__init__.py [hunk 2]
```diff
@@ -36,7 +36,7 @@
     'xthor', 'abnormal', 'scenetime', 'btdigg', 'transmitthenet', 'tvchaosuk',
     'torrentproject', 'extratorrent', 'bitcannon', 'torrentz', 'pretome', 'gftracker',
     'hdspace', 'newpct', 'elitetorrent', 'bitsnoop', 'danishbits', 'hd4free', 'limetorrents',
-    'norbits', 'ilovetorrents', 'sceneelite', 'anizb', 'bithdtv', 'zooqle'
+    'norbits', 'ilovetorrents', 'sceneelite', 'anizb', 'bithdtv', 'zooqle', 'animebytes',
```
Comment (medariox) [inline sickbeard/providers/__init__.py]: Maybe remove the last comma for consistency.

**Comment by p0psicles** [comment]
Also rebased on dev

**Comment by medariox** [comment]
Approved

**Pull request merged**